### ISSUE TITLE
coqui_stt_training.train never finishes

### ISSUE THREAD

**Issue by FrontierDK** [coqui-ai/STT#2195]
Hi all.

I am having an issue where coqui_stt_training.train never finishes. Even if I try using --epochs 3, it just waits forever for "something".

![Python1](https://user-images.githubusercontent.com/25281742/163978331-ee5a11d9-196f-4147-b4b6-8b76bef2f761.png)

This is where it is just waiting. CPU usage has dropped to 0% - and stays there.
![Python2](https://user-images.githubusercontent.com/25281742/163978332-085201fe-e3ee-496f-a253-3ab8acacbdce.png)

Is there any way of seeing what it is waiting for?

**Comment by HarikalarKutusu** [comment]
We replicated the issue. I confirm the issue is existing on Colab, and so does @wasertech (ran on local machine).

**Comment by wasertech** [comment]
Yep check out the full logs here: https://gist.github.com/wasertech/1cbde2c1399148cf9954fc42d272265b.

**Comment by HarikalarKutusu** [comment]
I checked with pip (v1.3.0), git clone v1.3.0 and git clone main all failed.
Could not pass another bug with v1.2.0
But it worked OK with git clone v1.1.0

**Comment by reuben** [comment]
Can you provide a script to reproduce locally? Are you using GPUs/CUDA? Does this only happen on Collab?

**Comment by wasertech** [comment]
I can confirm it hangs on GPU with Docker and inside Colab too. I think @FrontierDK was using CPU but I'm not sure.
Run 3 epochs on any dataset with any batch size using `python -m coqui_stt_training.train` and weather you are using only the training interface or in conjonction with the testing one, the process never retunrns any exit code after done.

**Comment by FrontierDK** [comment]
> I think @FrontierDK was using CPU but I'm not sure.

I was/am using CPU - true :)

**Comment by wasertech** [comment]
So I try to use the old train interface to make the same test and it still doesn't return any exit code.  https://gist.github.com/wasertech/a7bd3ae2606e143bf70a540972c3314b

**Comment by wasertech** [comment]
@reuben #2198 is a script to reproduce locally. Just run the script with some arguments like: 
```bash
path_to_mailabs="/mnt/extracted/data/M-AILABS/"
mailabs_lang="fr_FR"
path_to_alphabet="/mnt/models/alphabet.txt"
path_to_scorer="/mnt/lm/kenlm.scorer"
./bin/run-ci-mailabs_time.sh ${path_to_mailabs} ${mailabs_lang} ${path_to_alphabet} ${path_to_scorer}
```

As you can see in my [logs](https://gist.github.com/wasertech/543cd54e59079a6764aae98dfc977737), it never reaches [the lines bellow 39](https://github.com/wasertech/STT/blob/62c2600702547a7a508a450869a0d6a05102ab85/bin/run-ci-mailabs_time.sh#L40-L46) to print the exec time.

**Comment by wasertech** [comment]
I managed to fix it in [this branch](https://github.com/wasertech/STT/tree/fix-exit-train) by [commenting out the memory test](https://github.com/coqui-ai/STT/pull/2199/files#diff-8eac0cf5056d600145fb17a2c8449ea6d95c73979df7aaeb4b46d488a7997b20).

**Comment by wasertech** [comment]
So with #2205 you'll be able to skip the memory test which is causing this issue by using `coqui_stt_training.train` with the flag `--skip_batch_test` set to `true`.

**Comment by FrontierDK** [comment]
When I try using  _--skip_batch_test=true_ I get _train.py: error: unrecognized arguments: --skip_batch_test=true_

**Comment by wasertech** [comment]
@FrontierDK Try with `--skip_batch_size true` instead.

**Comment by FrontierDK** [comment]
My line
python3 -m coqui_stt_training.train --checkpoint_dir ~/coqui-stt-1.3.0-checkpoint --train_files ~/speech/sample.csv --dev_files ~/speech/dev.csv --test_files ~/speech/test.csv --n_hidden 2048 --load_cudnn false --epochs 3 **--skip_batch_size true**

Then I get _train.py: error: unrecognized arguments: --skip_batch_size true_

**Comment by FrontierDK** [comment]
I don't think it has merged yet...I just tried installing a new VM and I still get this error:
_train.py: error: unrecognized arguments: --skip_batch_size_

This should have gotten the latest changes into the VM:
_git clone --depth 1 --recurse-submodules --shallow-submodules https://github.com/coqui-ai/STT_

**Comment by HarikalarKutusu** [comment]
Hmmm. Common Voice v9.0 is out and I want to start training some languages. It seems I will be working from a patched fork :(

**Comment by wasertech** [comment]
@FrontierDK  It is merged. I've just rebuilt a docker container from main and tested the following:
```zsh
❯ docker build --rm -f Dockerfile.train -t stt-train .
...
Successfully tagged stt-train:latest

❯ docker run -it --entrypoint 'bash' stt-train
root@3412bf80716e:/code# cat VERSION
1.4.0-alpha.1
root@3412bf80716e:/code# python -m coqui_stt_training.train --help
...
--skip_batch_test true/false
                        Coqpit Field: skip batch size memory test before
                        training
...
```
As expected. [Full logs](https://gist.github.com/wasertech/70e641c96a801713a81c389fe8755d07)

@HarikalarKutusu Why? It doen't work for you too?

**Comment by HarikalarKutusu** [comment]
@wasertech 👍  Thanks for the heads up, I didn't try yet, just assumed that it is not merged.

**Comment by FrontierDK** [comment]
I just installed a new/fresh VM, still seeing the error here.

As yesterday, Coqui was installed with _git clone --depth 1 --recurse-submodules --shallow-submodules https://github.com/coqui-ai/STT_

**Comment by hammondm** [comment]
Has this change been made with the docker image yet?

**Comment by HarikalarKutusu** [comment]
I could get it with git clone main repo...

**Comment by wasertech** [comment]
> Has this change been made with the docker image yet?
> 
> this: ghcr.io/coqui-ai/stt-train:latest
> 
> Update: the pre-built image is not yet updated but, like the person below, I was able to rebuild it from `Dockerfile.train` in the latest repo.

In deed, docker images are released on GitHub’s Docker repositories only for stable versions. Since STT 1.4.0 is still currently in alpha, there is no published image yet as the codebase is still subject to changes before the final release.

**Comment by wasertech** [comment]
> Has this change been made with the docker image yet?
> 
> this: ghcr.io/coqui-ai/stt-train:latest
> 
> Update: the pre-built image is not yet updated but, like the person below, I was able to rebuild it from `Dockerfile.train` in the latest repo.

Well I found [this image](https://github.com/coqui-ai/STT/pkgs/container/stt-train/20747534?tag=597e6ebb5c1a398c10b3a84810f93c1ce96d7926) which should contain the latest fixes but it doesn’t bare the ‘latest’ tag.

**Comment by reuben** [comment]
latest means latest stable release. We tag builds from main with the `main` tag as well: https://github.com/coqui-ai/STT/pkgs/container/stt-train/20747534?tag=main

**Comment by bernardohenz** [comment]
I'm facing the same issue: training just hangs out at the end of training.

I've tested with the following docker images:
- ghcr.io/coqui-ai/stt-train:latest
- ghcr.io/coqui-ai/stt-train:597e6ebb5c1a398c10b3a84810f93c1ce96d7926
- ghcr.io/coqui-ai/stt-train:e7d2af96bd16556c7ef828a660acc881dd51c4b
- docker building from master

All of them finished training but keep hanging there :(

**Comment by wasertech** [comment]
Yes, @bernardohenz. But I as I've said in my comments above:
Since STT 1.4.0, ...
>..., you can now use the —skip_batch_size flag when running coqui_stt_training.train module.

This will skip the memory test that create this issue.

**Comment by bernardohenz** [comment]
Thanks @wasertech , I didn't get at the first, sorry :+1:
